dms/models/directory.py

```diff
@@ -272,6 +272,71 @@ def check_access(self, operation, raise_exception=False):
             res = True
         return res
 
+    allowed_model_ids = fields.Many2many(
+        compute="_compute_allowed_model_ids", comodel_name="ir.model", store=False
+    )
+    model_id = fields.Many2one(
+        comodel_name="ir.model",
+        domain="[('id', 'in', allowed_model_ids)]",
+        compute="_compute_model_id",
+        inverse="_inverse_model_id",
+        string="Model",
+        store=False,
+    )
+
+    res_model = fields.Char(string="Res model")
+
+    res_id = fields.Integer(string="Res id")
+
+    record_ref = fields.Reference(
+        string="Record Referenced",
+        compute="_compute_record_ref",
+        selection=[],
+        readonly=True,
+        store=False,
+    )
+
+    storage_id_save_type = fields.Char(
+        compute="_compute_storage_id_save_type", store=False
```

Comment:
This can be a related field:
```suggestion
        related="storage_id.save_type", store=False
```